components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -265,6 +267,291 @@
+        }
+        return identityProviderList;
+    }
+
+    /**
+     * Get number of IdP count for a matching filter.
+     *
+     * @param tenantId       Tenant Id of the identity provider.
+     * @param expressionNode filter value list for IdP search.
+     * @return number of IdP count for a given filter
+     * @throws IdentityProviderManagementException Error when getting count of Identity Providers.
+     */
+    int getCountOfFilteredIdPs(int tenantId, List<ExpressionNode> expressionNode)
+            throws IdentityProviderManagementException {
+
+        String sqlStmt = IdPManagementConstants.SQLQueries.GET_IDP_COUNT_SQL;
+        int countOfFilteredIdp = 0;
+        FilterQueryBuilder filterQueryBuilder = new FilterQueryBuilder();
+        appendFilterQuery(expressionNode, filterQueryBuilder);
+        Map<Integer, String> filterAttributeValue = filterQueryBuilder.getFilterAttributeValue();
+        sqlStmt = sqlStmt + filterQueryBuilder.getFilterQuery() +
+                IdPManagementConstants.SQLQueries.GET_IDP_COUNT_SQL_TAIL;
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement prepStmt = dbConnection.prepareStatement(sqlStmt)) {
+            for (Map.Entry<Integer, String> prepareStatement : filterAttributeValue.entrySet()) {
+                prepStmt.setString(prepareStatement.getKey(), prepareStatement.getValue());
+            }
+            prepStmt.setInt(filterAttributeValue.entrySet().size() + 1, tenantId);
+            try (ResultSet rs = prepStmt.executeQuery()) {
+                if (rs.next()) {
+                    countOfFilteredIdp = Integer.parseInt(rs.getString(1));
+                }
+            }
+        } catch (SQLException e) {
+            throw new IdentityProviderManagementException("Error occurred while retrieving Identity Provider count " +
+                    "for a tenant : " + IdentityTenantUtil.getTenantDomain(tenantId), e);
+        }
+        return countOfFilteredIdp;
+    }
+
+    /**
+     * Create a sql query and prepared statement for filter.
+     *
+     * @param expressionNodes    list of filters.
+     * @param filterQueryBuilder Sql builder object.
+     */
+    private void appendFilterQuery(List<ExpressionNode> expressionNodes, FilterQueryBuilder filterQueryBuilder) {
+
+        StringBuilder filter = new StringBuilder();
+        if (CollectionUtils.isEmpty(expressionNodes)) {
+            filterQueryBuilder.setFilterQuery(IdPManagementConstants.EMPTY_STRING);
+        } else {
+            for (ExpressionNode expressionNode : expressionNodes) {
+                String operation = expressionNode.getOperation();
+                String value = expressionNode.getValue();
+                String attributeName = expressionNode.getAttributeValue();
+                if (StringUtils.isNotBlank(attributeName) || StringUtils.isNotBlank(value) || StringUtils
```

Comment:
shouldn't this be 
if (StringUtils.isNotBlank(attributeName) && StringUtils.isNotBlank(value) && StringUtils.isNotBlank(operation))